packages/mattermost-redux/src/selectors/entities/channel_categories.ts

```diff
@@ -104,7 +105,7 @@ export function makeFilterAutoclosedDMs(): (state: GlobalState, channels: Channe
         getCurrentUserId,
         getMyChannelMemberships,
         getChannelMessageCounts,
-        (state: GlobalState) => getInt(state, Preferences.CATEGORY_SIDEBAR_SETTINGS, Preferences.LIMIT_VISIBLE_DMS_GMS, 20),
+        (state: GlobalState, channels, categoryType: string, defaultDmLimit?: number) => getInt(state, Preferences.CATEGORY_SIDEBAR_SETTINGS, Preferences.LIMIT_VISIBLE_DMS_GMS, defaultDmLimit ?? 20),
```

Comment:
Is there any better way to pass `defaultDmLimit`

Reply:
I think, if we're going to keep doing it this way, we make a reusable selector in `mattermost-redux/selectors/entities/preferences` that includes the logic to check the current user's create_at timestamp to figure out the default. That'll save us from repeating that logic elsewhere, and it'll mean that the parameters for `filterAutoclosedDMs` don't need to be changed since the limit will still be calculated automatically by it